src/Analysis/Codelyzer.Analysis/CodeGraph.cs

```diff
@@ -0,0 +1,198 @@
+                .Select(n => n.Identifier)
+                .Distinct()
+                .ToHashSet();
+
+            // All class declarations. These will represent class nodes
+            var allClassDeclarations = allSourceFileResults.SelectMany(s => s.AllClasses());
+
+            var graph = new List<Node>();
+            allClassDeclarations.ToList().ForEach(classDeclaration => graph.Add(new Node()
+            {
+                Name = classDeclaration.Identifier,
+                Identifier = classDeclaration.FullIdentifier,
+                NodeType = NodeType.Class
+            }));
+
+            var interfaceDeclarations = allSourceFileResults.SelectMany(s => s.AllInterfaces());
+            interfaceDeclarations.ToList().ForEach(interfaceDeclaration => graph.Add(new Node()
+            {
+                Name = interfaceDeclaration.Identifier,
+                Identifier = interfaceDeclaration.FullIdentifier,
+                NodeType = NodeType.Interface
+            }));
+
+            foreach (var classDeclaration in allClassDeclarations)
+            {
+                var currentNode = graph.FirstOrDefault(n => n.Identifier == classDeclaration.FullIdentifier);
+
+                // Inheritance Relations (Interfaces or other classes)
+                var baseTypes = classDeclaration.BaseList;
+                baseTypes.ForEach(baseType =>
+                {
+                    var targetNode = graph.FirstOrDefault(t => t.Identifier == baseType);
+                    var edge = new Edge()
+                    {
+                        EdgeType = EdgeType.Inheritance,
+                        TargetNode = targetNode,
+                        SourceNode = currentNode
+                    };
+                    currentNode.Edges.Add(edge);
+                    targetNode.Edges.Add(edge);
+                });
+
+                // Declarations
+                var declarations = classDeclaration.AllDeclarationNodes().Where(d => allClassDeclarations.Any(c => c.FullIdentifier == d.FullIdentifier)).ToList();
+                declarations.ForEach(declaration =>
+                {
+                    var targetNode = graph.FirstOrDefault((n) => n.Identifier == declaration.FullIdentifier);
+                    if (targetNode.Identifier != currentNode.Identifier)
+                    {
+                        var edge = new Edge()
+                        {
+                            EdgeType = EdgeType.Declaration,
+                            TargetNode = targetNode,
+                            SourceNode = currentNode
+                        };
+                        currentNode.Edges.Add(edge);
+                        targetNode.Edges.Add(edge);
+                    }
+                });
+
+                // Method Calls
+                var invocationExpressions = classDeclaration.AllInvocationExpressions().Where(i => allNamespaces.Contains(i.SemanticNamespace)).ToList();
```

Comment:
does this include constructor ?

Reply:
Yes - ObjectCreationExpressions are when constructors are invoked, and they are a type of invocation expressions.